include/pinocchio/multibody/joint/joint-common-operations.hpp

```diff
@@ -51,25 +53,68 @@ namespace pinocchio
   {
     template<typename ConfigVectorIn, typename Scalar, typename ConfigVectorOut>
     static void run(
-      const Eigen::MatrixBase<ConfigVectorIn> & q,
+      const Eigen::MatrixBase<ConfigVectorIn> & qIn,
       const Scalar & scaling,
       const Scalar & offset,
-      const Eigen::MatrixBase<ConfigVectorOut> & dest)
+      const Eigen::MatrixBase<ConfigVectorOut> & qOut)
     {
-      assert(q.size() == dest.size());
-      PINOCCHIO_EIGEN_CONST_CAST(ConfigVectorOut, dest).noalias() =
-        scaling * q + ConfigVectorOut::Constant(dest.size(), offset);
+      assert(qIn.size() == qOut.size());
+      PINOCCHIO_EIGEN_CONST_CAST(ConfigVectorOut, qOut).noalias() =
+        scaling * qIn + ConfigVectorOut::Constant(qOut.size(), offset);
+    }
+  };
+
+  struct UnboundedRevoluteAffineTransform
+  {
+    template<typename ConfigVectorIn, typename Scalar, typename ConfigVectorOut>
+    static void run(
+      const Eigen::MatrixBase<ConfigVectorIn> & qIn,
+      const Scalar & scaling,
+      const Scalar & offset,
+      const Eigen::MatrixBase<ConfigVectorOut> & qOut)
+    {
+      assert(qIn.size() == 2);
+      assert(qOut.size() == 2);
+
+      const typename ConfigVectorIn::Scalar & ca = qIn(0);
+      const typename ConfigVectorIn::Scalar & sa = qIn(1);
+
+      const typename ConfigVectorIn::Scalar & theta = math::atan2(sa, ca);
+      const typename ConfigVectorIn::Scalar & theta_transform = scaling * theta + offset;
+
+      ConfigVectorOut & dest_ = PINOCCHIO_EIGEN_CONST_CAST(ConfigVectorOut, qOut);
+      SINCOS(theta_transform, &dest_.coeffRef(1), &dest_.coeffRef(0));
+    }
+  };
+
+  struct NoAffineTransform
+  {
+    template<typename ConfigVectorIn, typename Scalar, typename ConfigVectorOut>
+    static void run(
+      const Eigen::MatrixBase<ConfigVectorIn> & qIn,
+      const Scalar & scaling,
+      const Scalar & offset,
+      const Eigen::MatrixBase<ConfigVectorOut> & qOut)
+    {
+      assert(
+        check_expression_if_real<Scalar>(
+          math::fabs(scaling - 1.0) < Eigen::NumTraits<Scalar>::dummy_precision())
+        && check_expression_if_real<Scalar>(
+          math::fabs(offset) < Eigen::NumTraits<Scalar>::dummy_precision())
+        && "No ConfigVectorAffineTransform specialized for this joint type");
```

Comment:
I don't understand why there is these check.

Reply:
For joint where it is not trivial to define an affine transformation of the configuration, we still allow mimicking only if $\alpha = 1, \beta = 0$ (the identity transform: $q_{mimicked} = q_{mimicking}$)

Reply:
After discussion we must set an assert